data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -348,9 +348,18 @@ public void run() {
         boolean retryingAfterException = false;
         while (!shutdownInProgress.get()) {
             try {
-                if (retryingAfterException || pauseConsumePredicate.pauseConsuming()) {
-                    LOG.debug("Pause consuming from Kafka topic.");
+                if (retryingAfterException) {
+                    LOG.debug("Pause consuming from Kafka topic due a previous exception.");
                     Thread.sleep(10000);
+                } else if (pauseConsumePredicate.pauseConsuming()) {
+                    LOG.debug("Pause and skip consuming from Kafka topic due to an external condition: {}", pauseConsumePredicate);
+                    paused = true;
+                    consumer.pause(consumer.assignment());
+                    Thread.sleep(10000);
+                    continue;
+                } else if(paused) {
+                    LOG.debug("Resume consuming from Kafka topic.");
+                    consumer.resume(consumer.assignment());
```

Comment:
I guess you missed setting `paused = false` here

Reply:
Good catch. I just pushed a fix.